### ISSUE TITLE
improve bpftrace -l

### ISSUE THREAD

**Issue by brendangregg** [iovisor/bpftrace#20]
The current bpftrace -l implementation does partial matches only, and ditches wildcards. It should to wildcards properly, based on file globbing (not regular expressions).

For example, currently:

```
# ./src/bpftrace -l 'sleep*'
tracepoint:xfs:xfs_log_grant_sleep
tracepoint:sunrpc:rpc_task_sleep
tracepoint:block:block_sleeprq
tracepoint:compaction:mm_compaction_kcompactd_sleep
tracepoint:vmscan:mm_vmscan_kswapd_sleep
tracepoint:sched:sched_stat_sleep
tracepoint:syscalls:sys_enter_clock_nanosleep
tracepoint:syscalls:sys_exit_clock_nanosleep
[...]
```

That's probably not doing what people expect.

Since we have multiple fields colon delimited, it's a bit trickier than it might look. Here's my proposed solution:

"sleep\*" would match any field that begins with "sleep", followed by anything.

But you must also be able to include the colons in your search. So "tracepoint:\*" would list all tracepoints, and "syscalls:\*" would list all "tracepoint:syscalls:\*" plus anywhere else that "syscalls:" is a match. And "do_nano*" would match kprobe:do_nanosleep, etc. "do_nano" by itself would match nothing. "*nano*" would match anything that contains "nano".

**Comment by cneira** [comment]
@brendangregg 
I tried to improve probe listing using file globbing semantics, but it does not match your proposal. 
PR for this is https://github.com/iovisor/bpftrace/pull/193. Let me know if this works for you in this state or rework must be done to match your search pattern proposal.

**Comment by brendangregg** [comment]
fixed by #193
